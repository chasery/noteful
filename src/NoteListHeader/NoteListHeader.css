.NoteListHeader > h2 {
  margin: 0 0 1rem;
}

.NoteListHeader__folderControls {
  padding: 0 0 2.125rem;
}

.NoteListHeader__folderControls a + button {
  margin-left: 0.5rem;
}
