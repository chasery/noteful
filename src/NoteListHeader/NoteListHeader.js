import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter, Link } from "react-router-dom";
import NotefulContext from "../NotefulContext";
import "./NoteListHeader.css";

class NoteListHeader extends Component {
  static contextType = NotefulContext;

  handleDelete(folderId, callback) {
    const folderUrl = `http://localhost:8000/api/folders/${folderId}`;
    fetch(folderUrl, {
      method: "DELETE",
      "content-type": "application/json",
      mode: "cors",
    })
      .then((response) => {
        if (response.ok) {
          return response.status;
        } else {
          throw response.status;
        }
      })
      .then((json) => {
        this.props.history.push("/");
        callback(folderId);
      })
      .catch((status) => {
        console.log(status);
      });
  }

  render() {
    const { folderId } = this.props.match.params;
    const folder = this.context.folders.find(
      (folder) => folder.id === parseInt(folderId)
    );
    const folderHasNotes = this.context.notes.filter(
      (note) => note.folder_id === parseInt(folderId)
    );

    return (
      <>
        {folder ? (
          <div className="NoteListHeader">
            <h2>{folder.folder_name} Notes</h2>
            <div className="NoteListHeader__folderControls">
              <Link to={`/edit-folder/${folderId}`}>
                <button>Edit Folder</button>
              </Link>
              <button
                onClick={() =>
                  this.handleDelete(folder.id, this.context.deleteFolder)
                }
                disabled={folderHasNotes.length !== 0}
              >
                Delete Folder
              </button>
            </div>
          </div>
        ) : (
          <div className="NoteListHeader">
            <h2>All Notes</h2>
          </div>
        )}
      </>
    );
  }
}

NoteListHeader.propTypes = {
  context: PropTypes.shape({
    editFolder: PropTypes.func.isRequired,
    deleteFolder: PropTypes.func.isRequired,
  }),
};

export default withRouter(NoteListHeader);
