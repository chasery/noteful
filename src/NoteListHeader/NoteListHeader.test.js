import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import NoteListHeader from "./NoteListHeader";

describe("NoteListHeader component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<NoteListHeader />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
