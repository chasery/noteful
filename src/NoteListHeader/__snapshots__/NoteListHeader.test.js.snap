// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteListHeader component renders without error 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
